File: openHAB/UICircleButton.m
//
//  UICircleButton.m
//  openHAB
//
//  Created by Victor Belov on 03/04/14.
//  Copyright (c) 2014 Victor Belov. All rights reserved.
//

#import "UICircleButton.h"

@implementation UICircleButton

UIColor *normalBackgroundColor;
UIColor *normalTextColor;

- (id)initWithCoder:(NSCoder *)coder
{
//    NSLog(@"UICircleButton initWithCoder");
    self = [super initWithCoder:coder];
    if (self) {
        self.titleLabel.font = [UIFont fontWithName:@"HelveticaNeue-Light" size:20];
        self.layer.cornerRadius = self.bounds.size.width/2.0;
        self.layer.borderWidth = 1.0;
        self.layer.borderColor = self.titleLabel.textColor.CGColor;
        [self addTarget:self action:@selector(buttonActionReleased) forControlEvents:UIControlEventTouchUpInside];
        [self addTarget:self action:@selector(buttonActionTouched) forControlEvents:UIControlEventTouchDown];
        normalBackgroundColor = self.backgroundColor;
    }
    return self;
}

-(void)buttonActionReleased
{
}

-(void)buttonActionTouched
{
    self.backgroundColor = normalTextColor;
//    [self setTitleColor:normalBackgroundColor forState:UIControlStateNormal];
    normalTextColor = self.titleLabel.textColor;
    [self setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    [NSTimer scheduledTimerWithTimeInterval:0.2 target:self selector:@selector(timerTicked:) userInfo:nil repeats:NO];
}

-(void)timerTicked:(NSTimer*)timer
{
    self.backgroundColor = normalBackgroundColor;
    [self setTitleColor:normalTextColor forState:UIControlStateNormal];
    [timer invalidate];
}



@end

File: openHAB/UICircleButton.swift
// Copyright (c) 2010-2021 Contributors to the openHAB project
//
// See the NOTICE file(s) distributed with this work for additional
// information.
//
// This program and the accompanying materials are made available under the
// terms of the Eclipse Public License 2.0 which is available at
// http://www.eclipse.org/legal/epl-2.0
//
// SPDX-License-Identifier: EPL-2.0

import os.log
import UIKit

var normalBackgroundColor: UIColor?
var normalTextColor: UIColor?

class UICircleButton: UIButton {
    required init?(coder: NSCoder) {
        super.init(coder: coder)

        layer.borderWidth = 2
        layer.borderColor = UIColor(white: 0, alpha: 0.05).cgColor

        layer.cornerRadius = bounds.size.width / 2.0
    }
}
